pkg/project/v2/project_loader.go

```diff
@@ -194,19 +180,36 @@ func toEnvironmentSlice(environments map[string]manifest.EnvironmentDefinition)
 
 func loadProject(ctx context.Context, fs afero.Fs, loaderContext ProjectLoaderContext, projectDefinition manifest.ProjectDefinition, environments []manifest.EnvironmentDefinition) (Project, []error) {
 	if exists, err := afero.Exists(fs, projectDefinition.Path); err != nil {
-		return Project{}, []error{fmt.Errorf("failed to load project `%s` (%s): %w", projectDefinition.Name, projectDefinition.Path, err)}
+		formattedError := fmt.Errorf("failed to load project `%s` (%s): %w", projectDefinition.Name, projectDefinition.Path, err)
+		report.GetReporterFromContextOrDiscard(ctx).ReportLoading(report.StateError, formattedError, "", nil)
+		return Project{}, []error{formattedError}
 	} else if !exists {
-		return Project{}, []error{fmt.Errorf("failed to load project `%s`: filepath `%s` does not exist", projectDefinition.Name, projectDefinition.Path)}
+		formattedError := fmt.Errorf("failed to load project `%s`: filepath `%s` does not exist", projectDefinition.Name, projectDefinition.Path)
+		report.GetReporterFromContextOrDiscard(ctx).ReportLoading(report.StateError, formattedError, "", nil)
+		return Project{}, []error{formattedError}
 	}
 
 	log.Debug("Loading project `%s` (%s)...", projectDefinition.Name, projectDefinition.Path)
 
-	configs, errs := loadConfigsOfProject(ctx, fs, loaderContext, projectDefinition, environments)
-	errs = append(errs, findDuplicatedConfigIdentifiers(configs)...)
-	errs = append(errs, checkKeyUserActionScope(configs)...)
+	configs, errors := loadConfigsOfProject(ctx, fs, loaderContext, projectDefinition, environments)
+	for _, err := range errors {
+		report.GetReporterFromContextOrDiscard(ctx).ReportLoading(report.StateError, err, "", nil)
+	}
+	configsWithErrors := make(map[coordinate.Coordinate]error)
+	errors = append(errors, findDuplicatedConfigIdentifiers(configs, configsWithErrors)...)
+	errors = append(errors, checkKeyUserActionScope(configs, configsWithErrors)...)
```

Comment:
Would it theoretically be possible to have more than one error per config?

Reply:
I'll go for writing both